src/Lifi.ts

```diff
@@ -152,14 +148,12 @@ class LIFI {
 
       let stepExecutor: StepExecutor
       try {
-        stepExecutor = new StepExecutor()
-        this.activeRoutes[route.id].executors.push(stepExecutor)
-        await stepExecutor.executeStep(
-          signer,
-          step,
-          updateFunction,
-          this.activeRoutes[route.id].settings.switchChainHook
+        stepExecutor = new StepExecutor(
+          route,
+          this.activeRoutes[route.id].settings
         )
+        this.activeRoutes[route.id].executors.push(stepExecutor)
```

Comment:
Saving the Route inside the StepExecuter and the StepExecuter inside the Route could create some kind of recursive problems? 
Or is that the reason why there is a difference between `route` and `this.activeRoutes[route.id]`? If yes that's pretty implicit and hard to grasp

Reply:
Maybe we just store the executor in a map of routeIds?